### PR TITLE
store/gc_worker: Add log to record the reason why GC is not running

### PR DESCRIPTION
**Pull request by MyonKeminta**
### What problem does this PR solve? 

Weeks ago, a user found that GC stopped running for hours and then went back to working. They reported this issue to us and it took us some time to realize that it was caused by user's manual operation, though they didn't believe it at first.

GC work ticks periodically but in some conditions it wont trigger GC:
* GC is already running
* New calculated safe point is before the previous one (caused by enlarging GC lifetime)
* GC run interval haven't passed since last GC
* A GC has just finished within a minute ago

Due to that configurations of GC is recorded in a system table, which is edited by SQL queries, it's a little difficult to record how user changes them. But in this PR, the reason why GC is not started will be recorded to log. So if this kind of reports occurs again, we will be easier confirm if it was caused by user's operation.

### What is changed and how it works?

Added logs

### Check List

Tests

 - No code

Side effects

 - Increased code complexity

### CODE REVIEW

File: store/tikv/gcworker/gc_worker.go [hunk 1]
```diff
@@ -231,6 +232,7 @@ func (w *GCWorker) leaderTick(ctx context.Context) error {
 	// wait a while before starting a new job.
 	if time.Since(w.lastFinish) < gcWaitTime {
 		w.gcIsRunning = false
+		log.Infof("[gc worker] a gc job was skipped because another gc job has just finished")
```
Comment (ngaut) [inline store/tikv/gcworker/gc_worker.go]: Ditto

File: store/tikv/gcworker/gc_worker.go [hunk 2]
```diff
@@ -288,6 +290,7 @@ func (w *GCWorker) checkGCInterval(now time.Time) (bool, error) {
 	}
 
 	if lastRun != nil && lastRun.Add(*runInterval).After(now) {
+		log.Infof("[gc worker] a gc job was skipped because gc interval (%v) haven't past since last run (%v)", runInterval, lastRun)
```
Comment (ngaut) [inline store/tikv/gcworker/gc_worker.go]: Ditto

File: store/tikv/gcworker/gc_worker.go [hunk 3]
```diff
@@ -216,6 +216,7 @@ func (w *GCWorker) storeIsBootstrapped() bool {
 // Leader of GC worker checks if it should start a GC job every tick.
 func (w *GCWorker) leaderTick(ctx context.Context) error {
 	if w.gcIsRunning {
+		log.Infof("[gc worker] a gc job was skipped because there's already a gc job running")
```
[the remark this answers is not in the document]
Comment (MyonKeminta) [reply store/tikv/gcworker/gc_worker.go]: leaderTick is called every minute to check if it need to do GC. I think there's no more detail can be recorded.

[the remark this answers is not in the document]
Comment (ngaut) [reply store/tikv/gcworker/gc_worker.go]: Who am i? Who is a gc job? Any ID of mine Job and the "a gc job"? What's the current context?

**Comment by coocood** [comment]
@MyonKeminta 
We need to include the `uuid` of the GC worker in the log.

**Comment by coocood** [comment]
LGTM

**Comment by coocood** [comment]
@ngaut PTAL

**Pull request merged**